File: android/app/src/main/java/bjc/shoperp/service/restful/OrderGoodsService.java
package bjc.shoperp.service.restful;

import java.util.Date;
import java.util.HashMap;

import bjc.shoperp.domain.ColorFlag;
import bjc.shoperp.domain.OrderGoods;
import bjc.shoperp.domain.restfulresponse.domainresponse.GoodsCountCollectionResponse;

/**
 * Created by hcq on 2018/2/19.
 */

public class OrderGoodsService extends ServiceBase<OrderGoods> {

    public GoodsCountCollectionResponse GetGoodsCount(int[] flags, Date startTime, Date endTime, int pageIndex, int pageSize) throws Exception {
        HashMap<String, Object> para = new HashMap<>();
        para.put( "flags", flags );
        para.put( "startTime", startTime );
        para.put( "endTime", endTime );
        para.put( "pageIndex", pageIndex );
        para.put( "pageSize", pageSize );
        return DoPost( GoodsCountCollectionResponse.class, para, null );
    }
}

File: net/ShopErp.App/Service/Restful/OrderGoodsService.cs
using ShopErp.Domain;
using ShopErp.Domain.RestfulResponse;
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.IO;
using System.Linq;
using System.Windows;

namespace ShopErp.App.Service.Restful
{
    public class OrderGoodsService : ServiceBase<OrderGoods>
    {
        public DataCollectionResponse<GoodsCount> GetGoodsCount(ColorFlag[] flags, DateTime startTime, DateTime endTime,
            int pageIndex, int pageSize)
        {
            Dictionary<string, object> para = new Dictionary<string, object>();
            para["flags"] = flags;
            para["startTime"] = startTime;
            para["endTime"] = endTime;
            para["pageIndex"] = pageIndex;
            para["pageSize"] = pageSize;

            return DoPost<DataCollectionResponse<GoodsCount>>(para);
        }

        public DataCollectionResponse<SaleCount> GetSaleCount(long shopId, OrderType type, int timeType, DateTime startTime, DateTime endTime, int pageIndex, int pageSize)
        {
            Dictionary<string, object> para = new Dictionary<string, object>();
            para["shopId"] = shopId;
            para["type"] = type;
            para["timeType"] = timeType;
            para["startTime"] = startTime;
            para["endTime"] = endTime;
            para["pageIndex"] = pageIndex;
            para["pageSize"] = pageSize;
            return DoPost<DataCollectionResponse<SaleCount>>(para);
        }
    }
}